File: src/org/elixir_lang/debugger/node/command/RemoveBreakpoint.java
package org.elixir_lang.debugger.node.command;

import com.ericsson.otp.erlang.OtpErlangAtom;
import com.ericsson.otp.erlang.OtpErlangInt;
import com.ericsson.otp.erlang.OtpErlangObject;
import com.ericsson.otp.erlang.OtpErlangTuple;
import org.elixir_lang.debugger.node.Command;
import org.jetbrains.annotations.NotNull;

public class RemoveBreakpoint implements Command {
  private final String myModule;
  private final int myLine;

  public RemoveBreakpoint(@NotNull String module, int line) {
    myModule = module;
    myLine = line + 1;
  }

  @NotNull
  @Override
  public OtpErlangTuple toMessage() {
    return new OtpErlangTuple(new OtpErlangObject[] {
      new OtpErlangAtom("remove_breakpoint"),
      new OtpErlangAtom(myModule),
      new OtpErlangInt(myLine)
    });
  }
}

File: src/org/elixir_lang/debugger/node/command/RemoveBreakpoint.kt
package org.elixir_lang.debugger.node.command

import com.ericsson.otp.erlang.OtpErlangAtom
import com.ericsson.otp.erlang.OtpErlangInt
import com.ericsson.otp.erlang.OtpErlangTuple
import org.elixir_lang.debugger.node.Command

class RemoveBreakpoint(private val module: String, line: Int) : Command {
    private val line: Int = line + 1

    override fun toMessage(): OtpErlangTuple =
            OtpErlangTuple(arrayOf(OtpErlangAtom("remove_breakpoint"), OtpErlangAtom(module), OtpErlangInt(line)))
}
